codeforamerica/vita-min
# Nj 120 refund/balance due amounts

## Link to pivotal/JIRA issue

https://github.com/newjersey/affordability-pm/issues/120

## Is PM acceptance required?
- Yes - don't merge until JIRA issue is accepted!

**Reminder**: merge main into this branch and get green tests before merging to main

## What was done?
- I stubbed out several of the lines that get added into Lines 66 and 78, instead of leaving them out.
- I added `set_refund_owed_lines` in the calculator, which then populates the screens that tells the taxpayer about their refund/owed.
- Because the refund screen then asks about direct deposit/check, we now need default values in the intake mock for payment (I took the default values used multiple places [elsewhere](https://github.com/codeforamerica/vita-min/blob/70b369e398c846562eb7c95b373dee4af8d16a89/spec/lib/submission_builder/ty2024/states/id/id_return_xml_spec.rb#L5) in the repo)
- The one place I just left a comment is for calculating Line 80, where once we implement the inputs to Line 78 we will have to find some way to keep that value from going negative.
 
## How to test?
- Minimal profile, with $1000 already paid should mean they get a refund and see the first set of screenshots below
- Zeus many w2s should mean they owe a balance and see the second set of screenshots below

## Screenshots (for visual changes)

### Minimal with $1000 already paid

Refund screen:
<img width="1011" alt="image" src="https://github.com/user-attachments/assets/e74f1bff-1eaa-4545-ae3b-ffa85a883bde">

---

XML
<img width="517" alt="image" src="https://github.com/user-attachments/assets/30df3dfb-b256-4d4a-8a2c-42cdced88fa5">

---

PDF
<img width="678" alt="image" src="https://github.com/user-attachments/assets/2c4fc3be-9bb3-425b-8687-405a24370f3e">
<img width="656" alt="image" src="https://github.com/user-attachments/assets/2c75392a-fca7-4862-bad2-ce4bd129d5ed">

### Zeus Many W2s

Balance Owed Screen:

<img width="1017" alt="image" src="https://github.com/user-attachments/assets/f728b4e7-e138-4d04-a7b5-8bfd9307549f">

---

XML
<img width="468" alt="image" src="https://github.com/user-attachments/assets/a5a17fa9-3113-4d51-9c59-5627e9e8cce7">

---

PDF
<img width="664" alt="image" src="https://github.com/user-attachments/assets/3142c61d-1622-4403-8f12-5919cb18eafe">
<img width="648" alt="image" src="https://github.com/user-attachments/assets/4308f365-8f4d-4405-987e-e1895850cf77">
